cypher/cypher-docs/src/docs/dev/ql/administration/security/security-and-indexes-intro.asciidoc

```diff
@@ -0,0 +1,101 @@
+
+Consider blacklisting on the label `:Person`.
+
+    DENY TRAVERSE Person ON GRAPH * TO users;
+
+If the user runs a query that will use the native index:
+
+    MATCH (n:User) WHERE n.name CONTAINS 'ndy' RETURN n.name;
+
+This query will perform several checks:
+
+* do a scan on the index to create a stream of results of nodes with the `name` property, which leads to four results
+* filter the results to include only nodes where `n.name CONTAINS 'ndy'`, filtering out `Mark` and `Joe` so we have two results
+* filter the results to exclude nodes that also have the label `:Person`, filtering out `Mandy` so we only have one result
+
+For the above dataset, we can see we will get one result.
+
+What if we query this with the full-text index:
+
+    CALL db.index.fulltext.queryNodes("userNames", "ndy") YIELD node, score
+    RETURN node.name
+
+The problem now is that we do not know if the results provided by the index were because of a match to the `name` or the `surname` property.
+The steps taken by the query engine would be:
+
+* run a _Lucene_ query on the full-text index to produce results containing `ndy` in either property, leading to four results.
+* filter the results to exclude nodes that also have the label `:Person`, filtering out `Mandy` and `Joe` so we have two results.
+
+This difference in results is due to the `OR` relationship between the two properties in the index creation.
+
+==== Blacklisting properties
+
+Now consider blacklisting properties, like the `surname` property:
+
+    DENY READ {surname} ON GRAPH * TO users;
+
+Now we run the same queries again:
+
+    MATCH (n:User) WHERE n.name CONTAINS 'ndy' RETURN n.name;
+
+This query will operate exactly as before, returning the same single result, because nothing in this query relates to the blacklisted property.
+
+But consider the full-text index query:
+
+    CALL db.index.fulltext.queryNodes("userNames", "ndy") YIELD node, score
+    RETURN node.name
+
+The problem now is that we do not know if the results provided by the index were because of a match to the `name` or the `surname` property.
+Results from the surname need to be excluded by the security rules, because they require that the user cannot see any `surname properties.
+However, the security model is not able to introspect the _Lucene_ query to know what it will actually do, whether it works only on the allowed `name` property, or also on the disallowed `surname` property.
+We know that the earlier query returned a match for `Joe Andy` which should now be filtered out.
+So, in order to never result results the user should not be able to see, we have to block all results.
+The steps taken by the query engine would be:
+
+* Determine if the full-text index includes blacklisted properties
+* If yes, return an empty results stream, otherwise process as before
+
+The query will therefor return zero results in this case, rather than simply returning only the `Andy` result that might be expected.
```

Comment:
```suggestion
The query will therefore return zero results in this case, rather than simply returning only the `Andy` result that might be expected.
```